src/switchroot/ostree-system-generator.c

```diff
@@ -63,8 +63,8 @@ main (int argc, char *argv[])
    * exit so that we don't error, but at the same time work where switchroot
    * is PID 1 (and so hasn't created /run/ostree-booted).
    */
-  autofree char *ostree_target = get_ostree_target ();
-  if (!ostree_target)
+  autofree char *ostree_cmdline = read_proc_cmdline_key ("ostree");
```

Comment:
I suspect though that we may end up needing to care about aboot here too.  But if you don't think so, then this is definitely better.

What I was looking at was pushing all the cmdline parsing deep into the generator code instead of this binary so it can more cleanly use the internals, but it is a bigger change.

Reply:
Yup we have to be mindful of it (it works in the code as of today, but I understand there are big changes coming so we revisiting this will occur at times), stateroot being somewhat frozen in the cmdline is ok (you can theoretically change it, you just got to deliver a new Android Boot Image bundle in your update).... But with the other fields that's less ok. But I'm happy to change things as this code evolves and changes.

Reply:
And I also agree we should try and do this kind of thing in future:

```
+typedef enum {
+    OTCORE_BOOT_TYPE_OSTREE,
+    OTCORE_BOOT_TYPE_ANDROID,
+} OtCoreBootType;
+
+typedef struct {
+  OtCoreBootType type;
+  char *ostree_link;
+  char android_boot_slot;  // If type == ANDROID, this will be either 'a' or 'b'
+} OtCoreBootEntry;
```

If the differences start to get larger that 10 lines of code. At the moment, since it's just a single if statement that's called once in the lifetime of a process, it's not worth it.... yet... at least